esp/src/src/DataPatternsWidget.ts

```diff
@@ -49,6 +49,8 @@ type _TabContainerWidget = {
     addChild(tabItem: any, pos: number);
 };
 
+export const supportedFileType = (contentType: string): boolean => ["flat", "csv", "thor"].indexOf((contentType || "").toLowerCase()) >= 0;
```

Comment:
What about XML?

Reply:
The script submits a workunit containing generated ECL.  In that ECL is something like:

ds := DATASET(filePath, RECORDOF(filePath, LOOKUP), flat);

In other words, it relies on compile-time record definition lookup.  Is that available/correct with XML files?  My understand is that it was not (always) available and, if not, XML files should be ignored.

Reply:
@dcamper I didn't realize that was the issue - in theory I do have the actual RECORD available to me - if I used that, does it mean XML would also be supported (this is the ECL record as per the Logical File details page).

Reply:
DataPatterns.Profile() doesn't actually care about the underlying file format at all.  You can even profile a file on the landing zone with a properly-crafted DATASET pathname and record definition.  Bottom line, if you can determine the record layout then by all means, let's use it!  And please give me a hint offline, as I have some other code that needs modifying....

Reply:
FYI:  I have created a new Jira for the XML case.  As per a discussion I had with Gordon, this PR should be focused on only the undefined filetype case.